namesys/routing.go

```diff
@@ -30,15 +31,30 @@ func NewRoutingResolver(route routing.IpfsRouting) Resolver {
 
 // CanResolve implements Resolver. Checks whether name is a b58 encoded string.
 func (r *routingResolver) CanResolve(name string) bool {
-	_, err := mh.FromB58String(name)
+	log.Debugf("CanResolve: '%s'", name)
+	var p = path.FromString(name)
+
+	parts := p.Segments()
+	if parts[0] == "ipns" {
+		parts = parts[1:]
+	}
```

Comment:
not sure we want to make ipns path-aware. may be this is the right way to go, but let's justify first

- _why_ do we _need_ paths in the ipns resolvers?